### PR TITLE
atlauncher: fix build

### PR DESCRIPTION
**Pull request by GetPsyched**
## Description of changes

Launching any instance would throw an error due to a missing library.

Issues and solutions brought to light in https://discourse.nixos.org/t/atlauncher-executables-not-found

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GetPsyched** [comment]
Unsure if we really need the `udev` and `xorg.libXxf86vm` deps in the library path as I ran a Minecraft instance and it seemed to have run fine. Not removing those since I haven't done any exhaustive testing.

File: pkgs/by-name/at/atlauncher/package.nix [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+, lib
+, makeDesktopItem
+, makeWrapper
+, stdenv
+
+, # dependencies
+  libglvnd
+, udev
+, xorg
+}:
```
Comment (eljamm) [inline pkgs/by-name/at/atlauncher/package.nix:13]:
```suggestion

, gamemodeSupport ? stdenv.isLinux
, textToSpeechSupport ? stdenv.isLinux
, additionalLibs ? [ ]
}:

let
  runtimeLibs = [
    libglvnd
    libpulseaudio
    udev
    xorg.libXxf86vm
  ]
  ++ lib.optional gamemodeSupport gamemode.lib
  ++ lib.optional textToSpeechSupport flite
  ++ additionalLibs;
in

```

Taking inspiration from the [Prismlauncher wrapper](https://github.com/NixOS/nixpkgs/blob/1863d9cfeb0a36da5ff3a2896eb7ed9c7a7d6f77/pkgs/games/prismlauncher/wrapper.nix#L91), we can extract the necessary libs into one single `runtimeLibs`. Also, by adding an `additionalLibs` list, we can easily add new libraries without having to override the entire wrapper.

File: pkgs/by-name/at/atlauncher/package.nix [hunk 2]
```diff
@@ -25,7 +37,7 @@ stdenv.mkDerivation (finalAttrs: {
     cp $src $out/share/java/ATLauncher.jar
 
     makeWrapper ${jre}/bin/java $out/bin/atlauncher \
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ xorg.libXxf86vm udev ]}" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libglvnd udev xorg.libXxf86vm ]}" \
```
Comment (eljamm) [inline pkgs/by-name/at/atlauncher/package.nix:40]:
```suggestion
      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeLibs}" \
```

**Review by eljamm — COMMENTED** [review]
I think `udev` is necessary for Controller support, but I can't test it. As for `xorg.libXxf86vm`, my game ran fine without it, but it's probably fine to keep.

However, I think that a few improvements can still be made:

**Review by eljamm — APPROVED** [review]
Everything looks good to me.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**